base/src/dispatcher/dispatcher.ts

```diff
@@ -575,14 +579,24 @@ export class Dispatcher implements ZLUX.Dispatcher {
         }
         break;
       case ActionType.Minimize:
+          this.log.debug("invoke Launch, which means do nothing if wrapper found: "+wrapper);
+          this.windowManager.minimize(targetId);
+          console.log(targetId);
+          break;
       case ActionType.Maximize:
+         this.log.debug("invoke Launch, which means do nothing if wrapper found: "+wrapper);
+         this.windowManager.maximize(targetId);
         this.log.warn('Max/Min not supported at this time. Concern if apps should be able to control other apps visibility.');
+          break;
       default:
         this.log.warn("Unhandled action type = "+action.type);
       };
     });
   }
 
+  setWindowManager(windowManager:any) {
+    this.windowManager = windowManager;
```

Comment:
For security, I don't think the window manager should be able to be set more than once. Can this be `attachWindowManager` and return a boolean about if successful? The behavior would be if this.windowManager already exists, return false, else return true & assign.